frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,170 @@
+            collapsedText="Show More"
             addLabelControl={
               !isSavingEdits && (
                 <Label
-                  textMaxWidth="40ch"
+                  data-testid="add-label-button"
                   color="blue"
                   variant="outline"
-                  isOverflowLabel
-                  onClick={toggleAddLabelModal}
+                  isEditable
+                  editableProps={{
+                    'aria-label': 'Add label',
+                    defaultValue: '',
+                    'data-testid': 'add-label-input',
+                  }}
+                  onEditComplete={(_event, newText) => {
+                    const error = validateLabel(newText);
+                    if (error) {
+                      setLabelErrors((prev) => ({ ...prev, [newText]: error }));
+                    } else {
+                      setUnsavedLabels((prev) => [...prev, newText]);
+                      const newErrors = { ...labelErrors };
+                      delete newErrors[newText];
+                      setLabelErrors(newErrors);
+                    }
+                  }}
                 >
                   Add label
                 </Label>
               )
             }
           >
-            {unsavedLabels.map((label, index) => (
+            {unsavedLabels.map((label) => (
               <Label
-                key={label}
-                color="blue"
-                data-testid="label"
+                data-testid={`editable-label-${label}`}
+                key={`${label}-${labelErrors[label] ? 'error' : 'normal'}`}
+                color={labelErrors[label] ? 'red' : 'blue'}
                 isEditable={!isSavingEdits}
-                editableProps={{ 'aria-label': `Editable label with text ${label}` }}
                 onClose={() => removeUnsavedLabel(label)}
-                closeBtnProps={{ isDisabled: isSavingEdits }}
-                onEditComplete={(_event, newText) => {
-                  if (!reservedKeys.includes(newText) && newText.length <= 63) {
-                    editUnsavedLabel(newText, index);
-                  }
+                closeBtnProps={{
+                  isDisabled: isSavingEdits,
+                  'data-testid': `remove-label-${label}`,
+                }}
+                onEditComplete={(_event, newText) => handleEditComplete(_event, newText)}
+                editableProps={{
+                  defaultValue: '',
+                  'aria-label': 'Edit label',
+                  'data-testid': `edit-label-input-${label}`,
                 }}
               >
                 {label}
               </Label>
             ))}
           </LabelGroup>
+          {Object.keys(labelErrors).length > 0 && (
+            <Alert
+              data-testid="label-error-alert"
+              variant={AlertVariant.danger}
+              isInline
+              title={Object.values(labelErrors)[0]}
+              aria-live="polite"
+            />
```

Comment:
use `isPlain` prop to remove colored background.

Reply:
Got it, thank you!